api/webhooks/webhooks.py

```diff
@@ -43,97 +49,205 @@ class WebhookType(enum.Enum):
 }
 
 
-def get_webhook_model(webhook_type: WebhookType) -> typing.Union[WebhookModels]:
+def get_webhook_model(
+    webhook_type: WebhookType,
+) -> Type[Union[OrganisationWebhook, Webhook]]:
     if webhook_type == WebhookType.ORGANISATION:
         return OrganisationWebhook
     if webhook_type == WebhookType.ENVIRONMENT:
         return Webhook
 
 
-def call_environment_webhooks(environment, data, event_type):
+@register_task_handler()
+def call_environment_webhooks(
+    environment_id: int,
+    data: typing.Mapping,
+    event_type: str,
+    retries: int = WEBHOOK_BACKOFF_RETRIES,
+):
+    """
+    Call environment webhooks.
+
+    :param environment_id: The ID of the environment for which webhooks will be triggered.
+    :param data: A mapping containing the data to be sent in the webhook request.
+    :param event_type: The type of event to trigger the webhook.
+    :param retries: The number of times to retry the webhook in case of failure (int, default is 3).
+    """
+
     if settings.DISABLE_WEBHOOKS:
         return
-
+    try:
+        environment = Environment.objects.get(id=environment_id)
+    except Environment.DoesNotExist:
+        return
     _call_webhooks(
         environment.webhooks.filter(enabled=True),
         data,
         event_type,
         WebhookType.ENVIRONMENT,
+        retries,
     )
 
 
-def call_organisation_webhooks(organisation, data, event_type):
+@register_task_handler()
+def call_organisation_webhooks(
+    organisation_id: int,
+    data: typing.Mapping,
+    event_type: str,
+    retries: int = WEBHOOK_BACKOFF_RETRIES,
+):
+    """
+    Call organisation webhooks.
+
+    :param organisationt_id: The ID of the organisationt for which webhooks will be triggered.
+    :param data: A mapping containing the data to be sent in the webhook request.
+    :param event_type: The type of event to trigger the webhook.
+    :param retries: The number of times to retry the webhook in case of failure (int, default is 3).
+    """
+
     if settings.DISABLE_WEBHOOKS:
         return
-
+    try:
+        organisation = Organisation.objects.get(id=organisation_id)
+    except Organisation.DoesNotExist:
+        return
     _call_webhooks(
         organisation.webhooks.filter(enabled=True),
         data,
         event_type,
         WebhookType.ORGANISATION,
+        retries,
     )
 
 
 def call_integration_webhook(config, data):
-    return _call_webhook(config, data)
```

Comment:
Based on [this](https://github.com/Flagsmith/flagsmith/pull/2932/files#diff-86f132934c5b75bd30cb37ce30cc88b07b738d2b09d772cbeea42e2186809cbaR140), this looks wrong to me. The function still seems to expect us to send an ORM model, not an id.

Reply:
Yes, you are right. I missed changing this back.